### DISCUSSION TITLE
picture url in claim profile

### DISCUSSION THREAD

**Discussion by helmut72** [zitadel/zitadel#4631] [Ideas]
Would be cool if the profile picture will be uploaded to a S3 storage (or is provided by an own storage) and the picture url is included in claim 'profile'.
https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims

Users are lazy to upload their profile picture in every app. They are also confused if a company provides two places to maintain own profile pictures, one in AD/LDAP, the other for example in Zitadel.

@fforootd 
That's why I've asked [here](https://github.com/zitadel/zitadel/issues/4609#issuecomment-1291016140) to use the `jpegPhoto` from AD/LDAP to fill Zitadels profile picture, when Zitadel eventually supports picture url in claim 'profile' ;)

**Comment by fforootd** [comment]
Got it, makes sense.

For the assets we already have an [API](https://docs.zitadel.com/docs/apis/assets), but I think it is only available with an authorization. 

@livio-a @hifabienne I think we do not insert a URL into the claims at the moment, but this could be done by providing a random url for the asset with each token that gets sent out.

What do you think?

**Comment by hifabienne** [comment]
The url is already included in the profile. 
You need to configure on your applications, that you want to include the userinfo inside the id token and the picture should be included.

**Comment by helmut72** [comment]
Have tried it with Nextcloud, where OIDC plugin supports profile picture. Works! Great! Thanks!! :) But it's a steep learning curve for a OIDC/SAML noob like me...

**Comment by helmut72** [comment]
resolved
